index.test.js

```diff
@@ -0,0 +1,233 @@
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+
+  describe('match (capitalization doesnt matter)', () => {
+    it('should return true when inputs differ solely by capitalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.CAPITALIZATION);
+      const inputs = [['us', 'US']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: true });
+      });
+    });
+
+    it('should return false when inputs differ other than by capitalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.CAPITALIZATION);
+      const inputs = [['us', 'usa']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+
+  describe('match (whitespace doesnt matter)', () => {
+    it('should return true when inputs differ solely by shape of whitespaces', () => {
+      const instance = new Rubric().doesntMatter(Rubric.WHITESPACE_DIFFERENCES);
+      const inputs = [['the us of a', 'the	us   of\u2028\u2029a']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: true });
+      });
+    });
+
+    it('should return false when inputs differ other than by shape of whitespaces', () => {
+      const instance = new Rubric().doesntMatter(Rubric.WHITESPACE_DIFFERENCES);
+      const inputs = [['the us of a', 'The	us   of\u2028\u2029a']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+
+  describe('match (unicode normalization doesnt matter)', () => {
+    it('should return true when inputs differ solely by unicode normalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.UNICODE_NORMALIZATION);
+      // composed and decomposed forms of é
+      const inputs = [['\u00e9', '\u0065\u0301']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: true });
+      });
+    });
+
+    it('should return false when inputs differ other than by unicode normalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.UNICODE_NORMALIZATION);
+      // lowercase n \u006e and uppercase N \u004e with combining tilde \u0303
+      const inputs = [['\u006e\u0303', '\u004e\u0303']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+});
+
+describe('Real-world usage', () => {
+  describe('es rubric', () => {
+    const esRubric = new Rubric()
+      .doesntMatter(Rubric.UNICODE_NORMALIZATION)
+      .doesntMatter(Rubric.WHITESPACE_DIFFERENCES)
+      .doesntMatter(Rubric.CAPITALIZATION)
+      .doesntMatter(Rubric.es.COMMON_PUNCTUATION)
+      .doesntMatter(Rubric.es.COMMON_SYMBOLS)
+      .matters('-');
```

Comment:
Do you think we should leave `-` out of the set of punctuation? Maybe it's a little hard to qualify here, but it seems the hyphen should be considered differently from the em dash or en dash.

Reply:
My thinking was twofold. First, that hyphen is generally considered a punctuation character:
> The hyphen (‐) is a punctuation mark used to join words and to separate syllables of a single word.
from https://en.wikipedia.org/wiki/Hyphen#Unicode

As well, https://unicode-table.com/en/2010/ (hyphen) is in the Unicode "Punctuation marks" category. And personally, I'd be more surprised if hyphen was not included in a list of common punctuation than if it were.

And second, I liked that the two lists (common punctuation and common symbols) accounted for all the printable ASCII characters aside from `[ 0-0a-zA-Z]`, though this is more of a theoretical concern.

A complicating factor is that the character we generally use for hyphen, `-`, is 002d (hyphen-minus) which seems to be some sort of semantic mashup of both hyphen and the minus sign, my guess is because when Unicode was created, they observed that this is how this characters is widely used (in both roles) and so they rolled with it. It is not in the Punctuation marks category.

Nonetheless, I'd still personally feel surprised if it were not included in a list of common punctuation.

As for en-dash and em-dash, I didn't include either of them in common punctuation (via `COMMON_NONASCII_PUNCTUATION`), so at this point, they always matter. I'm open to whether they belong in `COMMON_NONASCII_PUNCTUATION`; common is obviously not a well-defined quality. The goal of what I'm trying to do here is that it shouldn't matter that much whether they're included or not; if you want them not to matter, it's easy to just do `.doesntMatter('–—')` 😁